routes/helpers/getMovieDetails.js

```diff
@@ -0,0 +1,49 @@
+/* dependencies */
+const async = require('async');
+const keystone = require('keystone');
+
+/* models */
+const Director = keystone.list('Director');
+const Theme = keystone.list('Theme');
+const Genre = keystone.list('Genre');
+
+let movieDetails;
+
+const getDirector = (cb) => {
+
+  Director.model.findOne({ _id: movieDetails.director, }).exec((err, director) => {
+    if (err) return cb(err);
+    return cb(null, director);
+  });
+};
+
+const getThemes = (cb) => {
+
+  Theme.model.find({ _id: { $in: movieDetails.themes, }, }).exec((err, themes) => {
+    if (err) return cb(err);
+    return cb(null, themes);
+  });
+};
+
+const getGenres = (cb) => {
+
+  Genre.model.find({ _id: { $in: movieDetails.genre, }, }).exec((err, genres) => {
+    if (err) return cb(err);
+    return cb(null, genres);
+  });
+};
+
+module.exports = (movie, cb) => {
+
+  movieDetails = movie;
+
+  async.parallel([ getDirector, getThemes, getGenres, ], (err, result) => {
+
+    if (err) return cb(err);
+    return cb(null, {
+      director: result[0],
```

Comment:
I'm struggling a bit with these more advanced bits of mongo db. But in any case, I think if I managed to get all of these things into one query it would end up being kind of unreadable and I feel like what I have (which I've now written better than what you've commented on above) is quite easy to understand.
If you can point me in the direction of a good example or something though then I'm happy to try again!